src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,327 @@
+        ),
+        default_value="SigmaClippingAggregator",
+        help="Name of the StatisticsAggregator subclass to be used.",
+    ).tag(config=True)
+
+    outlier_detector_list = List(
+        trait=Dict(),
+        default_value=None,
+        allow_none=True,
+        help=(
+            "List of dicts containing the name of the OutlierDetector subclass to be used, "
+            "the aggregated statistic value to which the detector should be applied, "
+            "and the validity range of the detector. "
+            "E.g. ``[{'apply_to': 'std', 'name': 'RangeOutlierDetector', 'validity_range': [2.0, 8.0]},]``."
+        ),
+    ).tag(config=True)
+
+    chunk_shift = Int(
+        default_value=None,
+        allow_none=True,
+        help=(
+            "Number of samples to shift the aggregation chunk for the calculation "
+            "of the statistical values. Only used in the second_pass(), since the "
+            "first_pass() is conducted with non-overlapping chunks (chunk_shift=None)."
+        ),
+    ).tag(config=True)
+
+    faulty_pixels_threshold = Float(
+        default_value=10.0,
+        allow_none=True,
+        help=(
+            "Threshold in percentage of faulty pixels over the camera "
+            "to identify regions of trouble."
+        ),
+    ).tag(config=True)
+
+    def __init__(
+        self,
+        subarray,
+        config=None,
+        parent=None,
+        stats_aggregator=None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        subarray: ctapipe.instrument.SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        stats_aggregator: ctapipe.monitoring.aggregator.StatisticsAggregator
+            The ``StatisticsAggregator`` to use. If None, the default via the
+            configuration system will be constructed.
+        """
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+
+        # Initialize the instances of StatisticsAggregator
+        self.stats_aggregator = {}
+        if stats_aggregator is None:
+            for _, _, name in self.stats_aggregator_type:
+                self.stats_aggregator[name] = StatisticsAggregator.from_name(
+                    name, subarray=self.subarray, parent=self
+                )
+        else:
```

Comment:
I guess you copied this style from `CameraCalibrator` or similar. But I am not sure we want to support this. I think it circumvents somewhat the configuration system and thus the provenance.